### PR TITLE
BizHawkClient: Improve updating component's suffix identifier

### PR DESCRIPTION
**Pull request by Zunawe**
## What is this fixing or adding?

Adds a new field for individual clients to specify the suffixes of their patch files. When the client is registered, the BizHawk Client component has its file identifier updated to include the new suffixes.

Also removes mentions of BizHawkClient from `LauncherComponents.py` entirely, which fixes some duplicated code. And now the component will _actually_ only load if a world imports it.

## How was this tested?

Tested with Emerald, and also made a dummy clients with each of `str`, `tuple[str, str]`, and `None` in their `patch_suffix` var, and then checked the listed file extensions in the "Open Patch" file selection dialog.

### CODE REVIEW

**Review by alwaysintreble — APPROVED** [review]
LGTM and once emerald has this the changes since my previous approval will also all be good

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**
